src/core/border_router/routing_manager.cpp

```diff
@@ -647,6 +648,9 @@ void RoutingManager::StartRouterSolicitationDelay(void)
     mTimeRouterSolicitStart = TimerMilli::GetNow();
     mRouterSolicitTimer.Start(randomDelay);
 
+    // Always stop stale check timer if Router Solicitation is in progress.
```

Comment:
The original intention is that since `ResetDiscoveredPrefixStaleTimer` will be re-called after RS is complete, it should be fine (or better) to stop the stale timer during RS.

But there should be no actual difference whether the stale timer is running or not since, as you said, there won't be two RS onging anyway.